File: 494-target-sum/494-target-sum.cpp
class Solution {
public:
    int cnt=0;
    int findTargetSumWays(vector<int>& nums, int target) {
        fun(nums,target,0,0);
        return cnt;
    }
    void fun(vector<int>& n,int t,int x,int sum){
        if(x==n.size()){
            if(sum == t) cnt++;
            return;
        }
        
        fun(n,t,x+1,sum+n[x]);
        fun(n,t,x+1,sum-n[x]);
    }
};

File: 494-target-sum/494-target-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var cnt=0;
var findTargetSumWays = function(nums, target) {
    cnt=0;
    fun(nums,target,0,0);
    return cnt;
};

function fun(n,t,sum,x){
    if(x===n.length){
        if(sum === t) cnt++;
        return;
    }
    
    fun(n,t,sum+n[x],x+1);
    fun(n,t,sum-n[x],x+1);
}